src/app/tracegraph/tracegraph.component.ts

```diff
@@ -9,15 +9,19 @@ import { Http } from '@angular/http';
 })
 
 export class TraceGraphComponent implements OnInit {
+    baseUri: string;
+    basePort: string;
 
     constructor( @Inject(ElementRef) private element: ElementRef, @Inject(Http) private http: Http) {
+        this.baseUri = process.env.ZIPKIN_HOST || 'localhost';
+        this.basePort = process.env.ZIPKIN_PORT || '9411';
     }
 
     ngOnInit() {
 
         this
             .http
-            .get(`http://localhost:9411/api/v1/dependencies?endTs=1474206961061&lookback=86400000`, {})
+            .get(`http://${this.baseUri}:${this.basePort}/api/v1/dependencies?endTs=1474206961061&lookback=86400000`, {})
```

Comment:
baseUri here is extremely misleading: a URI can already contain a port, so when reading this code I really would get confused. Given that you name the environment variables ZIPKIN_HOST and ZIPKIN_PORT, why rename them here?